lib/features/children/generosity_challenge/cubit/generosity_challenge_cubit.dart

```diff
@@ -101,6 +103,22 @@ class GenerosityChallengeCubit extends Cubit<GenerosityChallengeState> {
     _refreshState(days: newDays);
   }
 
+  Future<void> saveUserData(ChatScriptItem item) =>
+      _generosityChallengeRepository.saveUserData(
+        ChatScriptSaveKey.fromString(item.saveKey),
```

Comment:
Just the item.savekey will suffice as the first parameter if you fix the other comment. Then the ChatScriptSaveKey class can be deleted.